apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.cs

```diff
@@ -551,11 +583,20 @@ private async Task StopCompletionAsync(Task[] subscriberTasks, HashSet<Task> reg
             }
         }
 
+        /// <inheritdoc />
+        public override ValueTask DisposeAsync() => new ValueTask(StopAsync(_disposeTimeout));
+
         /// <inheritdoc />
         public override Task StopAsync(CancellationToken hardStopToken)
         {
             lock (_lock)
             {
+                // Check if the instance is started and shutdown is started/in-progress/faulted/completed. 
```

Comment:
I wonder whether it would be better to use `_globalSoftStopCts.IsCancellationRequested`, like for the Subscriber. (You'd want to move the call to `_globalSoftStopCts.Cancel() into the lock, potentially, to avoid race conditions.)

I'd be more comfortable with that as an indication that we've already called Stop - and it's then symmetrical with the subscriber. (Depending on WaitingForActivation feels a little implementation-specific.)

Reply:
Understood Jon. Done. (The idea was to check which of the two options would be preferable. 
I should have probably done those in the same file in two different commits)